builder/builder-git.c

```diff
@@ -56,6 +56,19 @@ git_get_mirror_dir (const char     *url_or_path,
   g_autoptr(GFile) git_dir = NULL;
   g_autofree char *filename = NULL;
   g_autofree char *git_dir_path = NULL;
+  GPtrArray *sources_dirs = NULL;
+  int i;
+
+  sources_dirs = builder_context_get_sources_dirs (context);
+  for (i = 0; i < sources_dirs->len; i++)
+    {
+      GFile* sources_root = g_ptr_array_index (sources_dirs, i);
+      g_autoptr(GFile) git_dir = g_file_get_child (sources_root, "git");
```

Comment:
I would be careful with reusing `git_dir` instead of shadowing it. I'd either use a different name for the variable to avoid `-Wshadow` warnings. Reusing the variable means that we will need to remember about clearing it before overwriting it later when we do not do an early return here or in the following iteration.